pkg/watcher/internal/metricsprovider/k8s.go

```diff
@@ -75,16 +75,16 @@ func NewMetricsServerClient() (watcher.MetricsProviderClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	return metricsServerClient{
+	return &metricsServerClient{
```

Comment:
What prompted this change?

- The struct has only 2 attributes so passing it by value does not incur a lot of copying
- A pointer has the disadvantage that the access to the attributes is indirect

It is not wrong but it has already been like this from the initial version, so why change it now?